dsd/constraints.py

```diff
@@ -576,6 +576,27 @@ class DomainPool(JSONSerializable):
 
     _idx: int = field(compare=False, hash=False, default=0, repr=False)
 
+    find_neighbor_sequences: bool = False
+
+    hamming_probability: Optional[Dict[int, float]] = field(default_factory=dict)
+    """
+    Dictionary that specifies probability of taking a new sequence from the pool that is some integer 
+    number of bases different. 
+    """
+
+    adjacency_list: Dict[str, List[str]] = field(default_factory=dict)
+    """
+    Dictionary storing used sequences and their neighbor sequences a certain Hamming distance away.
+    """
+
+    def __post_init__(self):
```

Comment:
Add full type annotations, e.g., 

```python
def __post_init__(self) -> None:
```

and

```python
def graph_hamming_distance(self, steps: int, previous_sequence: Optional[str]) -> Dict[str, List[str]]:
```

In fact, get mypy installed and run it so you can make sure that no new type errors are being introduced in your code. It should be set to warn about methods that do not have full type annotations on parameters and the return type.